From f93e1af51509f495cc2202230a650930863205fd Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 23 Jun 2021 17:18:24 +0300
Subject: [PATCH] [SW-46517] greco: implement load preboot function

For secureboot flow, ppboot uses arcboot ROM apis to perform
dma and verification of preboot image. This patch adds support
to use those APIs.

Change-Id: I6c491b6f46aa6cdd2d63d796eb2600b16cd4437e
---

diff --git a/arch/greco_arc/common.c b/arch/greco_arc/common.c
index 2de6bc9..4a27bf0 100644
--- a/arch/greco_arc/common.c
+++ b/arch/greco_arc/common.c
@@ -426,6 +426,12 @@
 	return PREBOOT_TEXT_BASE;
 }
 
+bool ppboot_load_preboot_img(u64 *loadaddr)
+{
+	*loadaddr = copy_preboot();
+	return true;
+}
+
 /*
  * This function makes sure no access is done from host to the SoC's PCIe IF.
  * The inbound memory requests are disabled. The ELBI access gets "retry"
